File: Bit Manipulation/67. Add Binary/67. Add Binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function (a, b) {
  a = BigInt("0b" + a);
  b = BigInt("0b" + b);

  while (b) {
    const without_carry = a ^ b;
    const with_carry = (a & b) << 1n;
    a = without_carry;
    b = with_carry;
  }

  return a.toString(2);
};

File: Bit Manipulation/67. Add Binary/67. Add Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        a, b = int(a, 2), int(b, 2)

        while b:
            without_carry = a ^ b
            with_carry = (a & b) << 1
            a, b = without_carry, with_carry
        
        return bin(a)[2:]

# Time: O(a + b)
# Space: O(1)